WIP: Fixes #170

danields761 opened the pull request:
As i described in #170, it's really annoying that this library hardly depends on sphinxcontrib-golangdomain and sphinxcontrib-dotnetdomain, so i made a fix attempt and leave it at your discretion

Changes:

* Makes sphinxcontrib-golangdomain and sphinxcontrib-dotnetdomain to be optional packages
* Adds check that appropriate dependencies of a specified api-type installed during the extension initialisation

AWhetter commented on autoapi/backends.py:
```diff
@@ -25,3 +25,35 @@
+}  # type: (Dict[str, Sequence[Tuple[str, str]]])
+
+
+def _get_available_backends():
```
Do you think it will simplify this if we check for the module in `sys.modules` instead (probably in `autoapi/extension.py` rather than here)? A user should have specified the domain extension in the `extensions` list in their `conf.py` already, so we can check that the extension is loaded.

danields761 replied:
That's is good point, will do in this way

AWhetter commented on autoapi/extension.py:
```diff
@@ -67,6 +69,23 @@ def run_autoapi(app):
+    # in reality if user will attempt to use `go` api type without
+    #  sphinxcontrib.golangdomain installed, sphinx will prohibit
+    #  all operations because of sphinxcontrib.golangdomain
+    #  extension is required
```
I don't think this comment is needed. You have made the error message nice and clear, so it is obvious what the code is checking without needing a comment.

danields761 commented:
@AWhetter, thank for your replies

I fixed raised remarks. Also i added "autoapi_type" setting check, which ensures this value is valid, as well as test-cases covering changes (using pytest, if you don't mind).

P.s. i see failed pipline, but don't see were i was wrong :)

AWhetter commented:
The linter is broken at the moment so don't worry about it.
Thanks for your change!

danields761 commented:
Thank you for merging this changes!

I think that this change should be covered inside documentation: optional dependencies available, ex: `pip install sphinx-autoapi[go,dotnet]`. But i don't think i can write it clear.
